src/charm.py

```diff
@@ -104,28 +106,31 @@ def _on_config_changed(self, _):
                 logger.info("Setting {} to: {}".format(key, value))
                 self._stored.config[key] = value
                 change_set.add(key)
-
-        patched_config = self._patch_config(self._stored.config)
-        change_set.update(patched_config)
-        self._stored.config.update(patched_config)
-
-        # use change set to support single dispatch of a config change.
-        template_change_set = {
-            "base-path",
-            "architecture",
-            "threads",
-            "http_proxy",
-            "https_proxy",
-            "mirror-list",
-        }
-        if len(change_set & template_change_set) > 0:
-            self._render_config(self._stored.config)
-        if "cron-schedule" in change_set:
-            if self._stored.config["cron-schedule"] == "":
-                self._remove_cron_job()
-            else:
-                self._setup_cron_job(self._stored.config)
-        self._update_status()
+        try:
+            patched_config = self._patch_config(self._stored.config)
+        except ValueError as err:
```

Comment:
**suggestion:** 

I will consider write down the possible error in the `self._patch_config` docstring.
This somehow make sure we are handle a `knowed issue`

Reply:
Done